src/apps/backend/modules/config/internals/config_manager.py

```diff
@@ -0,0 +1,68 @@
+import os
+from pathlib import Path
+from typing import Any, Optional
+
+from modules.config.internals.config_file import ConfigFile
+
+
+class ConfigManager:
+    """
+    Manages application configuration by loading, merging, and providing access to config values.
+    """
+
+    CONFIG_KEY_SEPARATOR: str = "."
+
+    def __init__(self) -> None:
+        self.config_store: dict[str, Any] = {}
+        self.config_path = self._get_base_directory(__file__) / "config"
+
+    def get(self, key: str, default: Optional[Any] = None) -> Optional[Any]:
+        """
+        Retrieve a configuration value using dot-separated keys.
+        """
+        value = self.config_store
+        for k in key.split(self.CONFIG_KEY_SEPARATOR):
+            if not isinstance(value, dict) or k not in value:
+                return default
+            value = value[k]
+        return value
+
+    def has(self, key: str) -> bool:
+        """
+        Check if a configuration key exists.
+        """
+        value = self.config_store
+        for k in key.split(self.CONFIG_KEY_SEPARATOR):
+            if not isinstance(value, dict) or k not in value:
+                return False
+            value = value[k]
+        return True
+
+    def load_config(self) -> None:
```

Comment:
As discussed over the call, we need to consider better encapsulation of file, config, derived classes for Default or Env specific files.